### PR TITLE
[test-case-reporting] Add error handling to GET endpoints

### LINKED ISSUE #948: [test-case-reporting] Add type errors for bad GET params
Part of #881

### PR DESCRIPTION
**Pull request by Rafer45**
Closes #948 and #949

### CODE REVIEW

File: test-case-reporting/app.ts [hunk 1]
```diff
@@ -57,36 +68,42 @@ app.get('/', (req, res) => {
...
+    handleError(error, res);
   }
 });
 
 app.get('/test-results/history/:testCaseId', async (req, res) => {
-  // TODO(#949): Add error handling to GET endpoints
-  const {testCaseId} = req.params;
-  const {json} = req.query;
+  try {
+    const {testCaseId} = req.params;
```
Comment (rcebulko) [inline test-case-reporting/app.ts:92]: nit: these can't throw errors, so let's move them out of the try/catch and have the try encompass as small a set of actions as possbile

File: test-case-reporting/app.ts [hunk 2]
```diff
@@ -32,10 +32,21 @@ function renderTestRunList(testRuns: Array<TestRun>): string {
   return `Rendering ${testRuns.length} test runs!`;
 }
 
+function handleError(error: Error, res: express.Response): void {
+  console.warn(error);
+  if (error instanceof TypeError) {
```
Comment (rcebulko) [inline test-case-reporting/app.ts:37]: nit: if you're using this for a catch-all catch block, it's possible for `TypeError`s to bubble up unrelated to pagination. The best thing to do here is probably to extend `Error` with our own `PaginationError` and check for that. I don't really think it's worth that effort though tbh.

Comment (Rafer45) [reply test-case-reporting/app.ts:37]: I think characterizing bad input as a `TypeError` makes sense -- this isn't only for pagination errors, it also catches bad `buildNumber`s and bad `testCaseId`s. I don't know if I'm being a bit naïve in assuming nothing else will cause `TypeError`s, though.

File: test-case-reporting/app.ts (hunk 1 above)
Comment (Rafer45) [reply test-case-reporting/app.ts]: Done.

File: test-case-reporting/app.ts (hunk 2 above)
Comment (rcebulko) [reply test-case-reporting/app.ts:37]: It is certainly possible some other part of the code could throw a typeerror, though using TS obviously makes that much less likely. Like I said, I don't think it's worth the effort to change, just wanted to acknowledge the risk

**Comment by Rafer45** [comment]
Prematurely squash-merged before full reviews (I thought this was #952), please let me know if you have any comments @estherkim & we can revert to address them.

**Pull request merged**